2002-01-25  Jakub Jelinek  <jakub@redhat.com>

	* g++.dg/other/redecl1.C: New test.

--- gcc/testsuite/g++.dg/other/redecl1.C.jj	Fri Jan 25 00:19:22 2002
+++ gcc/testsuite/g++.dg/other/redecl1.C	Fri Jan 25 00:23:33 2002
@@ -0,0 +1,10 @@
+// PR c++/5857
+// This testcase failed because during duplicate_decls the type was promoted
+// to int.
+
+// { dg-do compile }
+
+typedef char baz;
+extern const char foo[];
+const baz foo[] = "xyz";
+const char bar[] = "abc";
